clients/da-vinci-client/src/main/java/com/linkedin/davinci/helix/LeaderFollowerPartitionStateModel.java

```diff
@@ -120,27 +118,14 @@ public void onBecomeStandbyFromOffline(Message message, NotificationContext cont
       if (isRegularStoreCurrentVersion) {
         waitConsumptionCompleted(resourceName, notifier);
       }
-      updateLagMonitor(
-          message.getResourceName(),
-          heartbeatMonitoringService::addFollowerLagMonitor,
-          false,
-          messageToString(message));
+      heartbeatMonitoringService
+          .updateLagMonitor(message.getResourceName(), getPartition(), HeartbeatLagMonitorAction.SET_FOLLOWER_MONITOR);
     });
   }
 
   @Transition(to = HelixState.LEADER_STATE, from = HelixState.STANDBY_STATE)
   public void onBecomeLeaderFromStandby(Message message, NotificationContext context) {
     LeaderSessionIdChecker checker = new LeaderSessionIdChecker(leaderSessionId.incrementAndGet(), leaderSessionId);
-    /**
-     * We set up the lag monitor first for leader transitions because we want to monitor the amount of time
-     * where a slice doesn't have a replicating leader.  While this state transition executes, there should be no
-     * other leader in the slice.
-     */
-    updateLagMonitor(
```

Comment:
Instead of moving this to SIT and making SIT take a dependency on the lag monitor, can we just move it into the below function call in executeStateIngestion?  I guess it would depend on if promoteToLeader is synchronous or not.  If it is, then we can do that and it's just a line move.  If it's async, then don't worry about it.

Reply:
Actually, SIT already take the dependency on the lag monitor service, as it needs to record HB. The method below for state transition to me is an async thing. For example for leader transition, we will only put the request into CAQ and execute and wait in async fashion.